ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/TransactionSimulator.java

```diff
@@ -198,6 +179,10 @@ public TransactionSimulator(
 
     final Transaction transaction = transactionBuilder.build();
 
+    final ProtocolSpec protocolSpec = protocolSchedule.getByBlockNumber(header.getNumber());
+
+    final MainnetTransactionProcessor transactionProcessor =
```

Comment:
What's the impact of moving this from before, to after the transaction is built?

Reply:
no impact. I can move it back